instance/import.go

```diff
@@ -0,0 +1,187 @@
+
+	if err := func() error {
+		observations, err := strconv.ParseInt(insert, 10, 64)
+		if err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "update imported observations: failed to parse inserted_observations string to int"), logData)
+			return errs.ErrInsertedObservationsInvalidSyntax
+		}
+
+		if err = s.UpdateObservationInserted(instanceID, observations); err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "update imported observations: store.UpdateObservationInserted returned an error"), logData)
+			return err
+		}
+
+		return nil
+	}(); err != nil {
+		if auditErr := s.Auditor.Record(ctx, UpdateInsertedObservationsAction, audit.Unsuccessful, auditParams); auditErr != nil {
+			err = auditErr
+		}
+		handleInstanceErr(ctx, err, w, logData)
+		return
+	}
+
+	s.Auditor.Record(ctx, UpdateInsertedObservationsAction, audit.Successful, auditParams)
+
+	log.InfoCtx(ctx, "update imported observations: request successful", logData)
+}
+
+// UpdateImportTask updates any task in the request body against an instance
+func (s *Store) UpdateImportTask(w http.ResponseWriter, r *http.Request) {
+
+	defer request.DrainBody(r)
+
+	ctx := r.Context()
+	vars := mux.Vars(r)
+	instanceID := vars["instance_id"]
+	auditParams := common.Params{"instance_id": instanceID}
+	logData := audit.ToLogData(auditParams)
+	defer r.Body.Close()
+
+	updateErr := func() *taskError {
+		tasks, err := unmarshalImportTasks(r.Body)
+		if err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "failed to unmarshal request body to UpdateImportTasks model"), logData)
+			return &taskError{err, http.StatusBadRequest}
+		}
+
+		validationErrs := make([]error, 0)
+		var hasImportTasks bool
+
+		if tasks.ImportObservations != nil {
+			hasImportTasks = true
+			if tasks.ImportObservations.State != "" {
+				if tasks.ImportObservations.State != models.CompletedState {
+					validationErrs = append(validationErrs, fmt.Errorf("bad request - invalid task state value for import observations: %v", tasks.ImportObservations.State))
+				} else {
+					if err := s.UpdateImportObservationsTaskState(instanceID, tasks.ImportObservations.State); err != nil {
+						log.ErrorCtx(ctx, errors.WithMessage(err, "Failed to update import observations task state"), logData)
+						return &taskError{err, http.StatusInternalServerError}
+					}
+				}
+			} else {
+				validationErrs = append(validationErrs, errors.New("bad request - invalid import observation task, must include state"))
+			}
+		}
+
+		if tasks.BuildHierarchyTasks != nil {
+			hasImportTasks = true
+			var hasHierarchyImportTask bool
+			for _, task := range tasks.BuildHierarchyTasks {
+				hasHierarchyImportTask = true
+				if err := models.ValidateImportTask(task.GenericTaskDetails); err != nil {
+					validationErrs = append(validationErrs, err)
+				} else {
+					if err := s.UpdateBuildHierarchyTaskState(instanceID, task.DimensionName, task.State); err != nil {
+						if err.Error() == "not found" {
```

Comment:
is there a strongly typed error we can test for instead of a string comparison, e.g. `if err == ErrNotFound`, string comparisons can be slow